wocky/wocky-tls.c

```diff
@@ -562,148 +171,54 @@ wocky_tls_session_add_ca (WockyTLSSession *session,
     }
   else if (S_ISREG (target.st_mode))
     {
-      n = gnutls_certificate_set_x509_trust_file (session->gnutls_cert_cred,
-          ca_path, GNUTLS_X509_FMT_PEM);
+      certs = g_tls_certificate_list_new_from_file (ca_path, NULL);
+      n += g_list_length (certs);
+      cas = g_list_concat (cas, certs);
       DEBUG ("+ %s: %d certs from file", ca_path, n);
     }
+  g_object_set_data_full (G_OBJECT (conn), "wocky-ca-list", cas, free_cas);
 }
 
 void
 wocky_tls_session_add_crl (WockyTLSSession *session, const gchar *crl_path)
 {
-  int n = 0;
```

Comment:
This (now empty function) seems to be used in wocky-connector-test only, otherwise it could be an security issue. Might be a reason for the test failing, though.

Reply:
Right, that was also called out in original patch here https://bugs.freedesktop.org/show_bug.cgi?id=31447#c15